omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/VisitConfigurationController2_0.java

```diff
@@ -0,0 +1,196 @@
+	public void updateCurrentConfiguration(@RequestBody VisitConfiguration newConfiguration) {
+		Context.requirePrivilege(PrivilegeConstants.CONFIGURE_VISITS);
+		AdministrationService administrationService = Context.getAdministrationService();
+		VisitService visitService = Context.getVisitService();
+		SchedulerService schedulerService = Context.getSchedulerService();
+
+		// validate
+		if (newConfiguration.getEnableVisits() && StringUtils.isEmpty(newConfiguration.getEncounterVisitsAssignmentHandler())) {
+			throw new IllegalRequestException("Visit encounter handler cannot be empty");
+		}
+
+		updateEnableVisitsValue(administrationService, newConfiguration.getEnableVisits());
+
+		if (newConfiguration.getEnableVisits()) {
+			updateVisitEncounterHandlerValue(administrationService,
+					newConfiguration.getEncounterVisitsAssignmentHandler());
+		}
+		updateGetAutoCloseVisitsTaskStartedValue(schedulerService, newConfiguration.getStartAutoCloseVisitsTask());
+		updateVisitTypesToAutoCloseValue(administrationService, visitService, newConfiguration.getVisitTypesToAutoClose());
+	}
+
+	private Boolean getEnableVisitsValue(AdministrationService administrationService) {
+		String enableVisits = administrationService
+				.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS, Boolean.FALSE.toString());
+		return Boolean.parseBoolean(enableVisits);
+	}
+
+	private void updateEnableVisitsValue(AdministrationService administrationService, Boolean value) {
+		GlobalProperty enableVisitsGp = new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS,
+				Boolean.toString(value));
+		administrationService.saveGlobalProperty(enableVisitsGp);
+	}
+
+	private String getVisitEncounterHandlerValue(AdministrationService administrationService,
+			EncounterService encounterService) {
+		String visitEncounterHandler = administrationService.getGlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER);
+		for (EncounterVisitHandler visitHandler : getEncounterVisitHandlers(encounterService)) {
+			if (visitHandler.getClass().getName().equals(visitEncounterHandler)) {
+				return visitHandler.getClass().getName();
+			}
+		}
+		return null;
+	}
+
+	private Collection<EncounterVisitHandler> getEncounterVisitHandlers(EncounterService encounterService) {
+		return encounterService.getEncounterVisitHandlers();
+	}
+
+	private void updateVisitEncounterHandlerValue(AdministrationService administrationService, String value) {
+		GlobalProperty visitEncounterHandlerGp = new GlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER, value);
+		administrationService.saveGlobalProperty(visitEncounterHandlerGp);
+	}
+
+	private Boolean getAutoCloseVisitsTaskStartedValue(SchedulerService schedulerService) {
+		TaskDefinition autoCloseVisitsTaskStarted = schedulerService
+				.getTaskByName(OpenmrsConstants.AUTO_CLOSE_VISITS_TASK_NAME);
+
+		if (autoCloseVisitsTaskStarted != null) {
+			return autoCloseVisitsTaskStarted.getStarted();
+		}
+
+		return null;
```

Comment:
Shouldn't this be false?

Reply:
You're right. Resolved here: https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/aed4a77af9dee8a5b24e7d0438f555a6a1d3a312